Python_Scripts/script_regroup_models.py

```diff
@@ -0,0 +1,337 @@
+
+
+def _check_sedml_submodels(sedml_file, sbml_files,
+                           model_name, model_year, model_info):
+    """Checks which sbml models of one sedml group belong together"""
+    sedml_path = os.path.abspath(sedml_file)
+    sedml_doc = libsedml.readSedML(sedml_file)
+    sedml_task_models = [task.getModelReference() for task in sedml_doc.getListOfTasks()]
+
+    for sbml_file in sbml_files:
+        # we want to find a simulation task for this SBML model
+        # first, get the name of the file (identifier in SED-ML)
+        sbml_path = os.path.abspath(sbml_file)
+        sbml_file_short = (sbml_file.split('/')[-1]).split('.')[0]
+
+        # Find the first task for this model (and warn if no task was found)
+        try:
+            task_number = sedml_task_models.index(sbml_file_short)
+        except ValueError:
+            print('SBML file not found in list of tasks: ' + sbml_file_short)
+            return model_info
+
+        # get the corresponding simulation setting
+        current_sim = sedml_doc.getSimulation(sedml_doc.getTask(
+            task_number).getSimulationReference())
+        # remember simulation settings
+        out_start = current_sim.getOutputStartTime()
+        out_end = current_sim.getOutputEndTime()
+        n_timepoints = current_sim.getNumberOfPoints()
+
+        # get info about the SBML file
+        sbml_model = libsbml.readSBML(sbml_file).getModel()
+        if sbml_model is None:
+            # check for SBML model
+            print('No sbml model found for file' + sbml_file +
+                  ', in model name ' + model_name +
+                  ', model year ' + model_year)
+            return model_info
+        n_species = len(sbml_model.getListOfSpecies())
+        n_reactions = len(sbml_model.getListOfReactions())
+
+        # add model information to the list
+        model_info.append({
+            'name': model_name,
+            'year': model_year,
+            'n_species': n_species,
+            'n_reactions': n_reactions,
+            'sbml_path': sbml_path,
+            'sedml_path': sedml_path,
+            'regrouped_path': '',
+            'start_time': out_start,
+            'end_time': out_end,
+            'n_timepoints': 101, # for whatever reason, our ref trajectories only use 101 points...
+            'long_id': (model_name, model_year, n_species, n_reactions),
+            'short_id': '',
+            'sedml_task': task_number,
+        })
+
+    return model_info
+
+
+def adapt_and_save_models(model_info_df):
+    # create the folders
+    if not os.path.exists(DIR_MODELS_REGROUPED):
+        os.mkdir(DIR_MODELS_REGROUPED)
+
+    model_folders = list(set(model_info_df['short_id']))
+    for model_folder in model_folders:
+        if not os.path.exists(os.path.join(DIR_MODELS_REGROUPED, model_folder)):
+            os.mkdir(os.path.join(DIR_MODELS_REGROUPED, model_folder))
+    logfile = open('sedml_change.log', 'w')
+    logfile.close()
```

Comment:
is this necessary? usually log-files do not need to be handled expicitly.

Reply:
If you use logging, then no.
I just used barebones writing-to-text, because was too lazy to look into logging-api... should update logging here